### PR TITLE
Forbid stray tokens in call index

### PR DESCRIPTION
**Pull request by Szegoo**
Closes: #12954

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2209067

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2209307

**Comment by Szegoo** [comment]
@ggwpez PTAL.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2209885

File: frame/support/procedural/src/pallet/parse/call.rs [hunk 1]
```diff
@@ -93,6 +93,10 @@ impl syn::parse::Parse for FunctionAttr {
 			let call_index_content;
 			syn::parenthesized!(call_index_content in content);
 			let index = call_index_content.parse::<syn::LitInt>()?;
+			if !index.suffix().is_empty() {
+				let msg = format!("invalid suffix `{:?}` for number literal", index.suffix());
+				return Err(syn::Error::new(lookahead.error().span(), msg))
```
Comment (bkchr) [inline frame/support/procedural/src/pallet/parse/call.rs:98]:
```suggestion
				return Err(syn::Error::new(index.span(), msg))
```

**Review by bkchr — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210420

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210610

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210657

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210704

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210751

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210802

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210868

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2210921

File: frame/support/procedural/src/pallet/parse/call.rs [hunk 2]
```diff
@@ -93,6 +93,10 @@ impl syn::parse::Parse for FunctionAttr {
 			let call_index_content;
 			syn::parenthesized!(call_index_content in content);
 			let index = call_index_content.parse::<syn::LitInt>()?;
+			if !index.suffix().is_empty() {
+				let msg = format!("invalid suffix `{:?}` for number literal", index.suffix());
```
Comment (bkchr) [inline frame/support/procedural/src/pallet/parse/call.rs:97]:
```suggestion
				let msg = "Number literal must not have a suffix";
```

Comment (bkchr) [reply frame/support/procedural/src/pallet/parse/call.rs:97]: Maybe you have a better idea for the error message? But there is no valid suffix, so we should not write "Invalid"

Comment (Szegoo) [reply frame/support/procedural/src/pallet/parse/call.rs:97]: Yes, you are right, I did not choose the correct word here.

**Review by ggwpez — APPROVED** [review]
Thanks!

Just as a note, you can run the pallet ui tests locally instead of relying on the CI. Eg for your call index test:  
```bash
RUN_UI_TESTS=1 rustup run 1.65 cargo test -p frame-support-test -- pallet_ui trybuild=call_index_has_suffix.rs
```
And optional with `TRYBUILD=overwrite` to overwrite the stderr files.

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**